acestep/core/generation/handler/generate_music.py

```
@@ -325,12 +330,27 @@ def generate_music(
             if audio_error is not None:
                 return audio_error
 
-            # Cover/repaint/lego/extract: lock duration to source audio.
+            # Cover/repaint/lego/extract/edit: lock duration to source audio.
             if processed_src_audio is not None and task_type in (
-                "cover", "cover-nofsq", "repaint", "lego", "extract",
+                "cover", "cover-nofsq", "repaint", "lego", "extract", "edit",
             ):
                 audio_duration = processed_src_audio.shape[-1] / self.sample_rate
 
+            # Flow-edit guards: paired CFG ≈ 4× decoder forwards/step, so
```

Ruff (0.15.12)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)